lib/ReactViews/DataCatalog/DataCatalog.jsx

```diff
@@ -26,24 +26,20 @@ class DataCatalog extends Component {
     const unfilteredItems =
       isSearching &&
       catalogSearchProvider &&
-      searchState.catalogSearchResults?.results
-        ? searchState.catalogSearchResults.results.map(
+      searchState.catalogSearchProvider?.result.results
+        ? searchState.catalogSearchProvider.result.results.map(
             (result) => result.catalogItem
           )
         : this.props.items;
     const items = (unfilteredItems || []).filter(defined);
     const { t } = this.props;
+
     return (
       <ul className={Styles.dataCatalog}>
         {isSearching && catalogSearchProvider && (
           <>
             <label className={Styles.label}>{t("search.resultsLabel")}</label>
-            <SearchHeader
-              searchResults={searchState.catalogSearchResults}
-              isWaitingForSearchToStart={
-                searchState.isWaitingToStartCatalogSearch
-              }
-            />
+            <SearchHeader searchResult={catalogSearchProvider.result} />
```

Comment:
This should be `catalogSearchProvider.searchResult` instead (currently this results in app crash when searching from catalog explorer modal)